upgrade/upgrade-scripts/upgrade

```diff
@@ -379,6 +379,8 @@ function finalize() {
 
 	[[ -n "$UPGRADE_TYPE" ]] ||
 		die "variable UPGRADE_TYPE is not set; is upgrade in progress?"
+	[[ -n "$UPGRADE_BASE_VERSION" ]] ||
+		die "variable UPGRADE_BASE_VERSION is not set"
 
 	case "$UPGRADE_TYPE" in
```

Comment:
Does this change work properly if a FULL upgrade (without a call to finalize) is performed first, then a DEFERRED upgrade (with a call to finalize) is performed next? How about a DEFERRED, then FULL, then DEFERRED? Just wondering if there's any potential issues that may occur, if a FULL upgrade was attempted at some point in the past upgrade history of the engine.

Reply:
I can test these out as well